op-challenger/runner/runner.go

```diff
@@ -195,7 +210,12 @@ func (r *Runner) runAndRecordOnce(ctx context.Context, rlog log.Logger, runConfi
 }
 
 func (r *Runner) runOnce(ctx context.Context, logger log.Logger, name string, gameType gameTypes.GameType, prestateSource prestateFetcher, localInputs utils.LocalGameInputs, dir string) error {
-	provider, err := createTraceProvider(ctx, logger, metrics.NewTypedVmMetrics(r.m, name), r.cfg, prestateSource, gameType, localInputs, dir)
+	if r.vmTimeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, r.vmTimeout)
+		defer cancel()
+	}
+	provider, err := r.traceProviderCreator(ctx, logger, metrics.NewTypedVmMetrics(r.m, name), r.cfg, prestateSource, gameType, localInputs, dir)
```

Comment:
We should treat these timeouts as a major error - equivalent to an invalid status code. If the VM hangs forever (even if it's because of a temporary RPC error), we will be unable to respond to games in a timely manner and games will resolve incorrectly.  Assuming our RPCs are down for hours at a time, if we're hitting this timeout it's either because games are taking way longer than we expect (which is a problem) or there's a major bug in the vm that needs attention.

Right now I think this is just considered a failure to run which would likely be ignored.